### ISSUE TITLE
Mobile phone port

### ISSUE THREAD

**Issue by abelsaug** [cmusatyalab/openface#279]
Hi guys!
I'd like to port openface to mobile platforms (iOS/Android).

**Comment by iwantooxxoox** [comment]
Hi @abelsaug,
I'd love to do it too, have any progress? Have you made any progress?
I'm trying to port from the [Tensorflow](https://github.com/davidsandberg/facenet) implementation to iOS

**Comment by abelsaug** [comment]
Hey,

I am currently going through their architecture actually. This Tensorflow
implementation is pretty nice. Does it include the changes made for
openFace 2.0 like triplet loss function? I'm a bit busy right now but I'll
be picking it up soon anyway.

**Comment by KirilaMenski** [comment]
Hi! Is there any progress?

**Comment by abelsaug** [comment]
Yeah, I picked it up last week again. I am just comparing the codebase  between the two and benchmarking them. I'll probably go with the tensorflow implementation since it would be probably be easier to port to mobile tensorflow. The triplet loss function and other improvements in openface 2.0 is included
Looks like @iwantooxxoox has started work in his facenet repository. We could clone it or it's original depending on the progress by @iwantooxxoox .

**Comment by iwantooxxoox** [comment]
Hi guys,
I'm working on porting [Openface](https://github.com/cmusatyalab/openface) (Torch implementation) to Keras from layer to layer, since there is a pre-trained weights (nn4.small2.v1.t7) on this Openface project, and why Keras is my aim is running Openface on iOS device, iOS 11 supports converting Keras to CoreML which makes us easier to leverage GPU on iOS device.

Any thoughts welcome.
Vic

**Comment by abelsaug** [comment]
Won't you be tackling a solution for Android?

On Aug 8, 2017 11:43, "victor" <notifications@github.com> wrote:

Hi guys,
I'm working on porting Openface <https://github.com/cmusatyalab/openface>
(Torch implementation) to Keras from layer to layer, since there is a
pre-trained weights (nn4.small2.v1.t7) on this Openface project, and why
Keras is my aim is running Openface on iOS device, iOS 11 supports
converting Keras to CoreML which makes us easier to leverage GPU on iOS
device.

Any thoughts welcome.
Vic

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/cmusatyalab/openface/issues/279#issuecomment-320860623>,
or mute the thread

**Comment by iwantooxxoox** [comment]
@abelsaug 
I will, but not at this moment, probably after finishing the iOS version.

**Comment by abelsaug** [comment]
Great, let me know how I can contribute!

**Comment by tpj3duke** [comment]
@iwantooxxoox what's the status on porting to Keras?

**Comment by iwantooxxoox** [comment]
@tpj3duke it's done, please kindly check out my GitHub repository: https://github.com/iwantooxxoox/Keras-OpenFace
